gst/join/gstjoin.c

```diff
@@ -0,0 +1,1076 @@
+/* SPDX-License-Identifier: LGPL-2.1-only */
+/**
+ * Copyright (C) 2020 Gichan Jang <gichan2.jang@samsung.com>
+ */
+/**
+ * @file	gstjoin.c
+ * @date	10 Nov 2020
+ * @brief	Select the out that arrived first among the input streams
+ * @see		https://github.com/nnstreamer/nnstreamer
+ * @author	Gichan Jang <gichan2.jang@samsung.com>
+ * @bug		No known bugs except for NYI items
+ */
+/**
+ * SECTION:element-join
+ * @see_also #GstInputSelector
+ * @note A join has reduced and changed input-selector's function.
+ *
+ * Connect recently arrived buffer from N input streams to the output pad.
+ * N streams should not operate at the same time.
+ * For example, If one sinkpad is receiving buffer, the others should be stopped.
+ * <refsect2>
+ * <title>Example launch line</title>
+ * gst-launch-1.0 ... (input stream 0) ! join.sink_0 \
+ *                ... (input stream 1) ! join.sink_1 \
+ *                ... \
+                  ... (input stream N) ! join.sink_n \
+                  join name=join ! (arrived input stream) ...
+ * </refsect2>
```

Comment:
Please add that all the capabilities (input stream i and output stream) should be the same.

Reply:
In this case, should we stop the pipeline or just warn them?

Reply:
If JOIN cannot enforce all the caps equalized, it should fail at the negotiation phase.
However, during the negotiation phase, caps are not finalized (it may be flexible) and JOIN should try to equalize them.

Reply:
Check capabilities of the sink pads are added.
The caps are checked when they are requested.